velox/substrait/SubstraitParser.cpp

```diff
@@ -17,147 +17,80 @@
   switch (substraitType.kind_case()) {
-    case ::substrait::Type::KindCase::kBool: {
-      typeName = "BOOLEAN";
-      nullability = substraitType.bool_().nullability();
-      break;
-    }
-    case ::substrait::Type::KindCase::kI8: {
-      typeName = "TINYINT";
-      nullability = substraitType.i8().nullability();
-      break;
-    }
-    case ::substrait::Type::KindCase::kI16: {
-      typeName = "SMALLINT";
-      nullability = substraitType.i16().nullability();
-      break;
-    }
-    case ::substrait::Type::KindCase::kI32: {
-      typeName = "INTEGER";
-      nullability = substraitType.i32().nullability();
-      break;
-    }
-    case ::substrait::Type::KindCase::kI64: {
-      typeName = "BIGINT";
-      nullability = substraitType.i64().nullability();
-      break;
-    }
-    case ::substrait::Type::KindCase::kFp32: {
-      typeName = "REAL";
-      nullability = substraitType.fp32().nullability();
-      break;
-    }
-    case ::substrait::Type::KindCase::kFp64: {
-      typeName = "DOUBLE";
-      nullability = substraitType.fp64().nullability();
-      break;
-    }
-    case ::substrait::Type::KindCase::kString: {
-      typeName = "VARCHAR";
-      nullability = substraitType.string().nullability();
-      break;
-    }
-    case ::substrait::Type::KindCase::kBinary: {
-      typeName = "VARBINARY";
-      nullability = substraitType.string().nullability();
-      break;
-    }
+    case ::substrait::Type::KindCase::kBool:
+      return BOOLEAN();
+    case ::substrait::Type::KindCase::kI8:
+      return TINYINT();
+    case ::substrait::Type::KindCase::kI16:
+      return SMALLINT();
+    case ::substrait::Type::KindCase::kI32:
+      return INTEGER();
+    case ::substrait::Type::KindCase::kI64:
+      return BIGINT();
+    case ::substrait::Type::KindCase::kFp32:
+      return REAL();
+    case ::substrait::Type::KindCase::kFp64:
+      return DOUBLE();
+    case ::substrait::Type::KindCase::kString:
+      return VARCHAR();
+    case ::substrait::Type::KindCase::kBinary:
+      return VARBINARY();
     case ::substrait::Type::KindCase::kStruct: {
-      // The type name of struct is in the format of:
-      // ROW<type0,type1,ROW<type2>>...typen.
-      typeName = "ROW<";
-      const auto& sStruct = substraitType.struct_();
-      const auto& substraitTypes = sStruct.types();
-      for (int i = 0; i < substraitTypes.size(); i++) {
-        if (i > 0) {
-          typeName += ",";
-        }
-        typeName += parseType(substraitTypes[i])->type;
+      const auto& substraitStruct = substraitType.struct_();
+      const auto& structTypes = substraitStruct.types();
+      std::vector<TypePtr> types;
+      std::vector<std::string> names;
+      for (int i = 0; i < structTypes.size(); i++) {
+        types.emplace_back(parseType(structTypes[i]));
+        std::string fieldName = "col_" + std::to_string(i);
+        names.emplace_back(fieldName);
```

Comment:
nit: fmt::format("col_{}", i)

Reply:
Updated. Could you help review again? Thanks.